hadoop-yarn-project/hadoop-yarn/hadoop-yarn-common/src/main/java/org/apache/hadoop/yarn/client/DefaultRMFailoverProxyProvider.java

```diff
@@ -0,0 +1,82 @@
+package org.apache.hadoop.yarn.client;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.io.retry.DefaultFailoverProxyProvider;
+import org.apache.hadoop.ipc.RPC;
+import org.apache.hadoop.yarn.conf.YarnConfiguration;
+
+import java.io.IOException;
+import java.net.InetSocketAddress;
+
+/**
+ * An implementation of {@link RMFailoverProxyProvider} which does nothing in the
+ * event of failover, and always returns the same proxy object.
+ * This is the default non-HA RM Failover proxy provider. It is used to replace
+ * {@Link DefaultFailoveProxyProvider} which was used as Yarn default non-HA.
+ */
+public class DefaultRMFailoverProxyProvider<T>
```

Comment:
For consistency, we should name this DefaultNoHaRMFailoverProxyProvider

Reply:
Successfully refactored